### PR TITLE
Add note regarding validation of nuget packages

### PR DESCRIPTION
**Pull request by LordMike**
### Why:
I was troubleshooting why my package push failed, and I remembered the note in the docs stating that yuo could upload multiple packages with the same repository url - which made me think, maybe that field (or some other) was checked and found invalid.

I've [fixed this blunder in my repo](https://github.com/LordMike/MBW.Libraries.DeviceIOControlLib/commit/431e45a1820adf631bbdd00a8c57e1a33eeea0f5), and suddenly the package was pushed. The issue was tricky to troubleshoot, as the [package push step just got a](https://github.com/LordMike/MBW.Libraries.DeviceIOControlLib/runs/2212097028?check_suite_focus=true) `403` back: `warn : Resource not accessible by integration`.

Please also consider tweaking this error code - it'd be helpful to know that the token was valid, but the package failed some validation check - or that the token was valid, but didn't have the permissions for this particular repo (I pushed to another repo's "packages").

### Check off the following:
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
@LordMike Thanks so much for opening a PR! I'll get this triaged for review ⚡

File: content/packages/guides/configuring-dotnet-cli-for-use-with-github-packages.md [hunk 1]
```diff
@@ -78,6 +78,8 @@ If your instance has subdomain isolation disabled:
 
 {% data reusables.package_registry.package-registry-with-github-tokens %}
 
+Note that nuget packages *must* have their RepositoryUrl set to the current repository. The package feed will refuse the package if it is incorrect.
```
Comment (HonkingGoose) [inline content/packages/guides/configuring-dotnet-cli-for-use-with-github-packages.md:81]:
```suggestion
Note that NuGet packages *must* have their RepositoryUrl set to the current repository. The package feed will refuse the package if it is incorrect.
```

NuGet is a proper noun, so probably should be capitalized.

**Review by Afsaneh1361 — CHANGES_REQUESTED** [review]
Operating ( System)

 _Thinking Requirements
[Android]


@LordMike 


_@Afsaneh1361

File: content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md [hunk 2]
```diff
@@ -227,6 +227,10 @@ Using packages from {% data variables.product.prodname_dotcom %} in your project
   dotnet restore
   ```
 
+### Troubleshooting
+
+The NuGet packages *must* have their `RepositoryUrl` set to the current repository. The package feed will refuse the package if it is incorrect.
```
Comment (jmarlena) [inline content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md:232]:
```suggestion
Your NuGet package may fail to push if the `RepositoryUrl` in *.csproj* is not set to the expected repository .
```

**Review by jmarlena — DISMISSED** [review]
One small suggestion for added context but this looks ready to ship! 


Thanks for surfacing this issue. I'll share your idea of a troubleshooting error improvement with the packages engineering team. 😸

**Review by jmarlena — APPROVED** [review]
🚢

**Pull request merged**